java-manta-client/src/main/java/com/joyent/manta/client/MantaSeekableByteChannel.java

```diff
@@ -347,33 +338,36 @@ public synchronized void close() throws IOException {
      * @return HTTP response object
      * @throws IOException thrown when there are network problems connecting to the remote API
      */
-    protected synchronized CloseableHttpResponse connectOrGetResponse() throws IOException {
-        if (responseRef.get() != null) {
-            return responseRef.get();
+    protected synchronized MantaObjectInputStream connectOrGetResponse() throws IOException {
+        if (responseStream.get() != null) {
+            return responseStream.get();
         }
 
         final HttpUriRequest request = connectionFactory.get(path);
+        final MantaHttpHeaders headers = new MantaHttpHeaders();
 
         // Set byte range requested via HTTP range header
-        request.setHeader(HttpHeaders.RANGE, String.format("bytes=%d-", position));
+        headers.setRange(String.format("bytes=%d-", position));
```

Comment:
That is the intention. I believe that there should be a hanging HTTP connection open. It will progressively read the bytes over that channel. If the consumer wants to end early, the close() call will break the connection.